src/applications/disability-benefits/526EZ/actions/index.js

```diff
@@ -1,64 +1,132 @@
-        setTimeout(resolve, ms);
+export function checkITFRequest(dispatch, hasSavedForm) {
+  return apiRequest(
+    '/intent_to_file',
+    null,
+    ({ data }) => {
+      let status;
+      let expirationDate;
+      const itfList = data.attributes.intentToFile;
+      // If the user does not have any existing ITFs, set none status
+      if (!itfList || (Array.isArray(itfList) && itfList.length === 0)) {
+        status = PRESTART_STATUSES.none;
+      // If the user has existing ITFs, check for active ITFs
       } else {
-        const errorMessage = 'Network request failed';
-        onChange({
-          ITFStatus,
-          errorMessage
-        });
-        reject(errorMessage);
+        const activeList = itfList.filter(itf => itf.status === PRESTART_STATUSES.active);
+        // If the user doesn't have any active ITFs, set expired status
+        if (activeList.length === 0) {
+          status = PRESTART_STATUSES.expired;
+          expirationDate = getLatestTimestamp(itfList.map(itf => itf.expirationDate));
+          // If the user has an active ITF, set retrieved status
+        } else {
+          status = PRESTART_STATUSES.retrieved;
+          expirationDate = activeList[0].expirationDate;
+        }
       }
-    });
+      dispatch(setPrestartStatus(status, expirationDate));
+      return status;
+    },
+    ({ errors }) => {
+      const status = hasSavedForm ? PRESTART_STATUSES.notRetrievedSaved : PRESTART_STATUSES.notRetrievedNew;
+      Raven.captureMessage('vets_itf_check_failure');
+      dispatch(setPrestartStatus(status, errors));
+      return status;
+    }
+  );
+}
 
-    return delay(2000);
-  };
+export function submitITFRequest(dispatch, successStatus, errorStatus) {
+
+  return apiRequest(
+    '/intent_to_file/compensation',
+    { method: 'POST' },
+    ({ data }) => {
+      dispatch(setPrestartStatus(successStatus, data.attributes.intentToFile.expirationDate));
+      return successStatus;
+    },
+    ({ errors }) => {
+      Raven.captureMessage('vets_itf_submission_error');
+      dispatch(setPrestartStatus(errorStatus, errors));
+      return errorStatus;
+    }
+  );
 }
 
-//   apiRequest(
-//     `${intentToFileUrl}`,
-//     null,
-//     ({ data }) => {
-//       const ITFStatus = data.attributes.ITFStatus
-//       onChange({ ITFStatus });
-//       return ITFStatus === 'active';
-//     },
-//     ({ errors }) => {
-//       const errorMessage = 'Network request failed';
-//       onChange({
-//         ITFStatus,
-//         errorMessage
-//       });
-//       Raven.captureMessage(`vets_itf_error: ${errorMessage}`);
-//     }
-//   );
-// };
-// }
-/* eslint-enable */
+export function verifyIntentToFile(hasSavedForm) {
+  return async (dispatch) => {
+    let submitSuccessStatus; // eslint-disable-line no-unused-vars
+    let submitErrorStatus;
+    dispatch(setPrestartStatus(PRESTART_STATUSES.pending));
 
+    const existingITFStatus = await checkITFRequest(dispatch, hasSavedForm);
```

Comment:
We expect this to be a string (`return status;`), but if the error handler in `checkITFRequest` throws, wouldn't we get an error back instead?

Reply:
Good point, should we log these to sentry instead?

Reply:
It looks like apiRequest already logs the error, so no need to do so here. I'll remove the logging portion of this.